cmd/generate/main.go

```diff
@@ -8,59 +8,126 @@
+		configPath = flag.String("config-path", "", "The path of json config file.\n"+
+			"The file can contain directiveMapName, matchFuncName, matchFuncComment, filter, and override.\n"+
+			"They provide same functions as other arguments directive-map-name, match-func-name, match-func-comment, filter, and override.\n"+
+			"It will unmarsh to generator.GenerateConfig. (optional)")
 		directiveMapName = flag.String("directive-map-name", "", "Name of the generated map variable."+
-			" Normally it should start with lowercase to avoid export. (required)")
+			"Normally it should start with lowercase to avoid export. If this is provided, the directiveMapName in json config will be ignored.\n"+
+			"You should provide it here or in json config.")
 		matchFuncName = flag.String("match-func-name", "", "Name of the generated matchFunc."+
-			" Normally it should start with uppercase to export. (required)")
+			"Normally it should start with uppercase to export. If this is provided, the match-func-name in json config will be ignored.\n"+
+			"You should provide it here or in json config.")
 		matchFnComment = flag.String("match-func-comment", "", "The code comment for generated matchFunc."+
-			" You can add some explanations like which modules included in it. Normally it should start with match-func-name (optional)")
+			"You can add some explanations like which modules included in it. Normally it should start with match-func-name.\n"+
+			"If this is provided, the matchFuncComment in json config will be ignored. (optional)")
 		filterflags       filterFlag
 		directiveOverride override
 	)
 	flag.Var(&filterflags, "filter",
 		"A list of strings specifying the directives to exclude from the output. "+
-			"An example is: -filter directive1 -filter directive2... (optional)")
+			"An example is: -filter directive1 -filter directive2...\n"+
+			"If this is provided, the filter in json config will be igonored. (optional)")
 	flag.Var(&directiveOverride, "override",
 		"A list of strings, used to override the output. "+
 			"It should follow the format:{directive:bitmask00|bitmask01...,bitmask10|bitmask11...}"+"\n"+
 			"An example is -override=log_format:ngxHTTPMainConf|ngxConf2More,ngxStreamMainConf|ngxConf2More"+"\n"+
-			`To use | and , in command line, you may need to enclose your input in quotes, i.e. -override="directive:mask1,mask2,...". (optional)`)
+			`To use | and , in command line, you may need to enclose your input in quotes, i.e. -override="directive:mask1,mask2,...".`+
+			`If this is provided, the override in json config will be ignored. (optional)`)
 
 	flag.Parse()
 
-	if *sourceCodePath == "" {
-		log.Fatal("src-path can't be empty")
+	var rawConfig cmdConfig
+	if *configPath != "" {
+		f, err := os.Open(*configPath)
+		if err != nil {
+			log.Println(err)
+			return
+		}
+
+		defer f.Close()
+
+		if err := json.NewDecoder(f).Decode(&rawConfig); err != nil {
+			log.Println(err)
```

Comment:
I think you may want use `log.Fatal` which will call `os.Exit(1)` so that the command exits with a non-success error code. If the goal is to not prevent `f.Close()` from running then you could wrap the logic in a function that returns the error so that it can be logged with `log.Fatal` after the file is closed.

Either way, I think you want to make sure that the command does not exit with a `0` status code when there is a failure.

Reply:
> If the goal is to not prevent f.Close() from running then you could wrap the logic in a function that returns the error so that it can be logged with log.Fatal after the file is closed.

Yeah that's what I want. Cool idea, applied in latest commit.